lib/flock/router/hash_ring.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Flock.Router.HashRing do
+  @moduledoc """
+  Hash Ring implementation of the Flock.Router behavior
+  """
+  @behaviour Flock.Router
+
+  def new(), do: HashRing.new()
+
+  def nodes(ring), do: HashRing.nodes(ring) 
+
+  def add_nodes(ring, nodes), do: Enum.reduce(nodes, ring, &HashRing.add_node(&2, &1))
+
+  def remove_nodes(ring, nodes), do: Enum.reduce(nodes, ring, &HashRing.remove_node(&2, &1))
+
+  def get_node(ring, key) do
+    case HashRing.key_to_node(ring, key) do
+      {:error, {:invalid_ring, reason}} -> {:error, reason}
```

Comment:
is `{:invalid_ring, reason}` the only error that can occur? Otherwise you're going to pass the error as an `{:ok, error_tuple}`

Reply:
Yeah it's a weird detail of `libring`